src/components/comment/commentInput/CommentInput.tsx

```diff
@@ -1,38 +1,74 @@
-import React from 'react';
-import { Icon } from '../..';
-import { CommentInputFunctions } from '../../../types';
+import React, { useState } from 'react';
+import { ProfileImage } from '../..';
 import S from './CommentInput.styled';
+import { useComment } from '../../../hooks';
+import { useParams } from 'react-router-dom';
+import { useQueryClient } from '@tanstack/react-query';
 
-const CommentInput = ({
-	contents,
-	addComment,
-	onKeyPress,
-	onChangeHandler,
-	onClickInput,
-}: CommentInputFunctions) => {
+const CommentInput = () => {
 	const isLogin = true; // 임시 코드
+	const postComment = useComment();
+	const { id } = useParams();
+	const pageNum = Number(id);
+	const [contents, setContents] = useState<string>('');
+	const queryClient = useQueryClient();
 
 	const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
 		event.preventDefault();
 	};
 
+	const addComment = () => {
+		if (contents.trim() !== '') {
+			const comment = {
+				content: contents,
+				isParent: true,
+			};
+			setContents('');
+			postComment.mutate(
+				{ pageNum, comment },
+				{
+					onSuccess: () => {
+						queryClient.invalidateQueries({ queryKey: ['detailedPage'] });
+					},
+				}
+			);
+		}
+	};
+	const onChangeHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+		setContents(event.target.value);
+	};
+	const onClickInput = () => {};
+
+	const onKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
+		const target = event.currentTarget;
+		if (target.value.length !== 0 && event.key === 'Enter') {
+			event.preventDefault();
+			addComment();
+		}
+	};
+
 	return (
 		<S.CommentInput>
 			<form onSubmit={onSubmit}>
-				<article className='user-input'>
-					<section className='user-input__icon'>
-						<Icon />
+				<article className='wrapper'>
+					<section className='container-user__icon'>
+						<div>
+							<ProfileImage url='' nickname={'yeom'} size='2.31rem' />
+						</div>
+						<span>{'johnyeom24'}</span>
```

Comment:
```suggestion
                    <div>
                        <ProfileImage url={userInfo?.imageUrl} nickname={userInfo?.nickname} size='2.31rem' />
                    </div>
                    <span>{userInfo?.nickname}</span>
```
해당 부분 유저가 적용 안되있습니다!! 
```typescript
const userInfo = useRecoilValue(userState); // 상단에 적용
```
유저 정보 접근하실때는 제가 다 userState에 저장해놓고 있어서, 참고하시면 될 것 같습니당!☺️

Reply:
오!! 감사합니다!! 수정한거 적용했습니다!!